File: client/src/types/knowledgeBase.ts
// Knowledge Base types matching backend DTOs

export interface FieldDescription {
  field_name: string;
  description: string;
}

export interface TableDescription {
  table_name: string;
  description: string;
  field_descriptions: FieldDescription[];
}

export interface KnowledgeBaseResponse {
  chat_id: string;
  table_descriptions: TableDescription[];
  created_at: string;
  updated_at: string;
}

export interface UpdateKnowledgeBaseRequest {
  table_descriptions: TableDescription[];
}

File: backend/internal/apis/dtos/knowledge_base.go
package dtos

import "neobase-ai/internal/models"

// UpdateKnowledgeBaseRequest is the request body for updating KB descriptions.
type UpdateKnowledgeBaseRequest struct {
	TableDescriptions []models.TableDescription `json:"table_descriptions" binding:"required"`
}

// KnowledgeBaseResponse is the response for KB endpoints.
type KnowledgeBaseResponse struct {
	ChatID            string                    `json:"chat_id"`
	TableDescriptions []models.TableDescription `json:"table_descriptions"`
	CreatedAt         string                    `json:"created_at"`
	UpdatedAt         string                    `json:"updated_at"`
}
